commit daedec1a61706a1107cae782cf88e953c76d1a29
Author: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date:   2019-05-15T12:15:05-06:00 (5 months ago)
    
    MANTA-4277 Plumb through object id RPC parameter for createobject function for node-boray

diff --git a/lib/objects.js b/lib/objects.js
index 64b443e..abd9268 100644
--- a/lib/objects.js
+++ b/lib/objects.js
@@ -28,7 +28,7 @@ var rpc = require('./rpc');
 
 ///--- API
 
-function createObject(rpcctx, owner, bucket_id, name, content_length,
+function createObject(rpcctx, owner, bucket_id, name, object_id, content_length,
     content_md5, content_type, headers, sharks, props, vnode, callback) {
     var opts, log;
 
@@ -36,6 +36,7 @@ function createObject(rpcctx, owner, bucket_id, name, content_length,
     assert.string(owner, 'owner');
     assert.string(bucket_id, 'bucket_id');
     assert.string(name, 'name');
+    assert.string(object_id, 'object_id');
     assert.number(content_length, 'content_length');
     assert.string(content_md5, 'content_md5');
     assert.string(content_type, 'content_type');
@@ -50,6 +51,7 @@ function createObject(rpcctx, owner, bucket_id, name, content_length,
     var arg = { owner: owner,
                 bucket_id: bucket_id,
                 name: name,
+                id: object_id,
                 vnode: vnode,
                 content_length: content_length,
                 content_md5: content_md5,
@@ -78,8 +80,9 @@ function createObject(rpcctx, owner, bucket_id, name, content_length,
     });
 }
 
-function createObjectNoVnode(rpcctx, owner, bucket_id, name, content_length,
-    content_md5, content_type, headers, sharks, props, callback) {
+function createObjectNoVnode(rpcctx, owner, bucket_id, name, object_id,
+    content_length, content_md5, content_type, headers, sharks, props,
+    callback) {
     var opts, log;
 
     assert.object(rpcctx, 'rpcctx');
@@ -98,6 +101,7 @@ function createObjectNoVnode(rpcctx, owner, bucket_id, name, content_length,
     var args = [ owner,
                  bucket_id,
                  name,
+                 object_id,
                  content_length,
                  content_md5,
                  content_type,
